config-proxy/src/main/java/com/yahoo/vespa/config/proxy/filedistribution/CachedFilesMaintainer.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.config.proxy.filedistribution;
+
+import com.yahoo.io.IOUtils;
+import com.yahoo.log.LogLevel;
+import com.yahoo.vespa.filedistribution.FileDownloader;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static java.nio.file.Files.readAttributes;
+
+/**
+ * Deletes cached file references and url downloads that have not been used for some time
+ *
+ * @author hmusum
+ */
+class CachedFilesMaintainer implements Runnable {
+
+    private final static Logger log = Logger.getLogger(CachedFilesMaintainer.class.getName());
+
+    private static final File defaultUrlDownloadDir = UrlDownloadRpcServer.downloadDir;
+    private static final File defaultFileReferencesDownloadDir = FileDownloader.defaultDownloadDirectory;
+    private static final Duration defaultDurationToKeepFiles = Duration.ofDays(30);
+
+    private final File urlDownloadDir;
+    private final File fileReferencesDownloadDir;
+    private final Duration durationToKeepFiles;
+
+    CachedFilesMaintainer() {
+        this(defaultFileReferencesDownloadDir, defaultUrlDownloadDir, defaultDurationToKeepFiles);
+    }
+
+    CachedFilesMaintainer(File fileReferencesDownloadDir, File urlDownloadDir, Duration durationToKeepFiles) {
+        this.fileReferencesDownloadDir = fileReferencesDownloadDir;
+        this.urlDownloadDir = urlDownloadDir;
+        this.durationToKeepFiles = durationToKeepFiles;
+    }
+
+    @Override
+    public void run() {
+        deleteUnusedFiles(fileReferencesDownloadDir);
```

Comment:
You need to make sure this method never throws any exception or that will stop the  `ScheduledExecutorService`.

Reply:
Good point, fixed